api/core.py

```diff
@@ -101,10 +104,19 @@
     'additionalProperties': False
 }
 
-def _append_targets(targets, container, prefix, total_size, total_cnt, optional, data_path, attachments=True):
+def _append_targets(targets, container, prefix, total_size, total_cnt, optional, data_path, filter_=None):
     for f in container['files']:
-        if (not attachments or (type(attachments) == list and f['filename'] not in attachments)) and 'attachment' in f.get('tags', []):
-            continue
+        if filter_:
+            if not filter_['attachments'] and f['flavor'] == 'attachment':
+                continue
+            if not filter_['dicom'] and f['filetype'] == 'dicom':
+                continue
+            if not filter_['nifti'] and f['filetype'] == 'nifti':
+                continue
+            if not filter_['montage'] and f['filetype'] == 'montage':
+                continue
+            if not filter_['other'] and f['flavor'] != 'attachment' and f['filetype'] not in ['montage', 'nifti', 'dicom']:
+                continue
```

Comment:
Hmm, makes sense. Could you place that explanation into the code as a comment?

I'm not sure where best to place it, maybe up on line 86 as a comment next to its part of the json schema.